src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.AzureKeyVault;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal static class JsonConfigurationParser
+    {
+        public static bool IsJsonContentType(string contentType)
+        {
+            try
+            {
+                string acceptedMainType = "application";
+                string acceptedSubType = "json";
+
+                ContentType ct = new ContentType(contentType.Trim().ToLower());
+                var type = ct.MediaType;
+                if (Equals(type, FeatureManagementConstants.ContentType)
+                    || Equals(type, KeyVaultConstants.ContentType))
+                {
+                    return false;
+                }
+
+                if (type.Contains('/'))
+                {
+                    string mainType = type.Split('/')[0];
+                    string subType = type.Split('/')[1];
+                    if (Equals(mainType, acceptedMainType))
+                    {
+                        if (subType.Contains('+'))
+                        {
+                            var subTypes = subType.Split('+');
+                            if (Array.Exists(subTypes, x => x == acceptedSubType))
+                            {
+                                return true;
+                            }
+                        }
+                        else if (Equals(subType, acceptedSubType))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is ArgumentException || ex is FormatException || ex is NullReferenceException)
```

Comment:
>NullReferenceException [](start = 97, length = 22)

Instead of catching `NullReferenceException`, we can use null conditional checks.